### DISCUSSION TITLE
Building all examples without boost and without asio

### DISCUSSION THREAD

**Discussion by yanboti** [randombit/botan#3790] [Q&A] [answered]
Hi all,

I would like to build examples but keep running into errors with asio and boost. Is there a way to turn it off? I am able to see that the configuration file allows for enabling boost, but not to turn it off.
I'm running the following command for the config:
```
./configure.py 
-cc=gcc -cxxflags=-std=c++20 
--with-os-features=getentropy
--with-external-includedir=/usr/include
--with-external-includedir=/usr/include/x86-64-linux-gnu
--with-external-libdir=/usr/lib
--with-external-libdir=/usr/lib/x86-64-linux-gnu
--build-targets=static
--build-targets=shared
--build-targets=examples
```

I run into problems with asio, so I ran the following:
`cp src/lib/tls/asio/*.h build/include/botan`
This clears up the errors pertaining to asio, but I'm not able to do the same for boost.
The error message is 
```
In file included from src/examples/tls_stream_client.cpp:3:
build/include/botan/asio_stream.h:15:10: fatal error: boost/version.hpp: No such file or directory
```
Any help would be appreciated. Thank you in advance.

**Comment by reneme** [comment, marked as answer]
Similar to what @securitykernel [explained in your other discussion](https://github.com/randombit/botan/discussions/3789#discussioncomment-7443361), examples are currently built as all-or-nothing. The `examples` build target is currently not integrated with Botan's module system and simply assumes that all modules required are available.

With #3787, `./configure.py` will at least refuse to allow the `examples` build target without `--with-boost`. But for a minimized build one still runs into compile errors, as you pointed out. Initially, we added the build target mostly to have all examples build in CI as a cross check.

But frankly, now that they are part of the "official" build scripts, perhaps we should actually clean this up and only build examples that fit the build configuration selected by the user.
